### ISSUE TITLE
Gamepad not autoconfiguring; effectively no input

### ISSUE THREAD

**Issue by QuarkTheAwesome** [libretro/RetroArch#5935]
This is a bit of a weird one. Some very recent changes seem to have cause the Wii U port to stop autoconfiguring the Gamepad; which essentially locks out input in most cases.
Here's my bisection:

- bfeb9ff ([RPX](https://puu.sh/yJq5C/43d0011b2d.rpx)) - Works fine; autoconfigures Gamepad on an empty SD card (fresh install).
- b2d5fad - Doesn't work; too much broken to determine where it fits in here.
- b724952 ([RPX](https://puu.sh/yJq7X/d2c984faeb.rpx)) - master at time of writing. Doesn't work; Gamepad is never configured.

[Here's the differences between `bfeb9ff` and `b724952`](https://github.com/libretro/RetroArch/compare/bfeb9ff...b724952) - the only changes are in the config file parser; so I'd assume the issue is something to do with parsing autoconfig profiles.

I did once get the following in the log; though I haven't been able to make it happen again (all my other builds simply didn't mention input at all):
```
[INFO] [Autoconf]: no profiles found for WIIU Gamepad (0/0).
```
I built using devkitPPC r29-1 (gcc 6.3) on a Linux host. My Wii U is on 5.5.2, using CBHC.

cc: @Alcaro @twinaphex

**Comment by twinaphex** [comment]
@Alcaro this seems directly related to the config file shenanigans we had been experiencing yesterday. What a world of pain we entered into after opening the file in non-textmode eh...

**Comment by QuarkTheAwesome** [comment]
Looks like a fix came as 2f422a4, but based on it being called "revert" I assume it's not the ideal solution.

**Comment by fr500** [comment]
It's temporary / up for discussion,  I branched out master to keep the code
intact before reverting so it can be solved properly

**Comment by gblues** [comment]
I wonder if you guys hit the same macro bug I did.

In an effort to centralize the pad name string, I had replaced "WIIU Gamepad" with a define constant. And when I did that, I ran into the same issue described in the OP--the gamepad wouldn't configure. (Fortunately, the actual pad driver was still processing button presses, so the L+R+LS+RS reset combo worked)

I traced it to a buggy macro that was treating the define macro as a literal string instead of the string, so the matcher code was trying to find a a config literally named `PAD_NAME_WIIU_GAMEPAD`.

A fix for this macro bug was included in the PR that was merged a couple days ago (see commit 704cc61fd8cf55537c111bf649286de15ad0b82c).
